content/frontend/vue-apollo/8-subscriptions.md

```diff
@@ -0,0 +1,324 @@
+  reconnect: true,
+  connectionParams: {
+    authToken: localStorage.getItem(GC_AUTH_TOKEN)
+  }
+})
+
+const networkInterfaceWithSubscriptions = addGraphQLSubscriptions(
+  networkInterface,
+  wsClient
+)
+
+networkInterface.use([{
+  applyBatchMiddleware (req, next) {
+    if (!req.options.headers) {
+      req.options.headers = {}
+    }
+    const token = localStorage.getItem(GC_AUTH_TOKEN)
+    req.options.headers.authorization = token ? `Bearer ${token}` : null
+    next()
+  }
+}])
+
+const apolloClient = new ApolloClient({
+  networkInterface: networkInterfaceWithSubscriptions,
+  connectToDevTools: true
+})
+```
+
+</Instruction>
+
+
+You're instantiating a `SubscriptionClient` that knows the endpoint for the Subscriptions API. Notice that you're also authenticating the websocket connection with the user's `token` that you retrieve from `localStorage`.
+
+Now you need to replace the placeholder `__SUBSCRIPTION_API_ENDPOINT__ ` with the endpoint for the subscriptions API.
+
+<Instruction>
+
+To get access to this endpoint, open up a terminal and navigate to the directory where `project.graphcool` is located. Then type the `graphcool endpoints` command. Now copy the endpoint for the `Subscriptions API` and replace the placeholder with it.
+
+</Instruction>
+
+
+> The endpoints for the Subscription API generally are of the form: `wss://subscriptions.graph.cool/v1/<project-id>`.
+>
+> Notice that if your project is not running in the "default" Graphcool [region](https://blog.graph.cool/new-regions-and-improved-performance-7bbc0a35c880), you need to add the your project's region to the endpoint like so: `wss://subscriptions.ap-northeast-1.graph.cool/v1/<project-id>` (for regoin _Asia Pacific_) or `wss://subscriptions.us-west-2.graph.cool/v1/<project-id>` (for _US West_).
+
+
+### Subscribing to new Links
+
+For the app to update in realtime when new links are created, you need to subscribe to events that are happening on the `Link` type. There generally are three kinds of events you can subscribe to:
+
+- a new `Link` is _created_
+- an existing `Link` is _updated_
+- an existing `Link` is _deleted_
+
+First, you need to add a subscription to `src/constants/graphql.js`:
+
+<Instruction>
+
+Open `src/constants/graphql.js` and add the following subscription:
+
+```js(path=".../hackernews-vue-apollo/src/constants/graphql.js")
+export const NEW_LINKS_SUBSCRIPTION = gql`
+  subscription {
+    Link(filter: {
+      mutation_in: [CREATED]
+    }) {
+      node {
+        id
+        url
+        description
+        createdAt
+        postedBy {
+          id
+          name
+        }
+        votes {
+          id
+          user {
+            id
+          }
+        }
+      }
+    }
+  }
+`
+```
+
+</Instruction>
+
+You'll implement this subscription in the `LinkList` component since that's where all the links are rendered.
+
+<Instruction>
+
+Open `src/components/LinkList.vue` and add the following method:
```

Comment:
You should use `subscribeToMore` in the smart query directly, see [doc](https://github.com/Akryum/vue-apollo#subscribetomore)

Reply:
@Akryum, can I have more than one `subscribeToMore` on the same smart query? I tried adding two (one for `NEW_LINKS_SUBSCRIPTION` and one for `NEW_VOTES_SUBSCRIPTION`), but I get a `Duplicate key 'subscribeToMore'` error.

Reply:
Try with the newest version of vue-apollo :smile:

Reply:
It would look like:

```javascript
subscribeToMore: [
  { document: ..., ... },
  { document: ..., ... },
],
```

Reply:
Thanks, @Akryum! This works after upgrading, but now I'm seeing `this.$apollo` being `undefined` throughout my app.